### PR TITLE
libstd: Fix typos in doc

### PR DESCRIPTION
**Pull request by rick68**

### CODE REVIEW

File: src/libstd/path.rs [hunk 1]
```diff
@@ -1627,7 +1627,7 @@ impl<'a> From<Cow<'a, Path>> for PathBuf {
 
 #[stable(feature = "shared_from_slice2", since = "1.24.0")]
 impl From<PathBuf> for Arc<Path> {
-    /// Converts a Path into a Rc by copying the Path data into a new Rc buffer.
+    /// Converts a PathBuf into a Arc by moving the PathBuf data into a new Arc buffer.
```
Comment (varkor) [inline src/libstd/path.rs]:
```suggestion
    /// Converts a `PathBuf` into an `Arc` by moving the `PathBuf` data into a new `Arc` buffer.
```

File: src/libstd/path.rs [hunk 2]
```diff
@@ -1647,7 +1647,7 @@ impl From<&Path> for Arc<Path> {
 
 #[stable(feature = "shared_from_slice2", since = "1.24.0")]
 impl From<PathBuf> for Rc<Path> {
-    /// Converts a Path into a Rc by copying the Path data into a new Rc buffer.
+    /// Converts a PathBuf into a Rc by moving the PathBuf data into a new Rc buffer.
```
Comment (varkor) [inline src/libstd/path.rs]:
```suggestion
    /// Converts a `PathBuf` into an `Rc` by moving the `PathBuf` data into a new `Rc` buffer.
```

Comment (varkor) [reply src/libstd/path.rs]: It would be good to put backticks around the comment in `impl From<&Path> for Rc<Path>` too.

**Comment by Alexendoo** [comment]
Ping from triage: Have you had a chance to look at the review comments @rick68?

**Comment by JohnCSimon** [comment]
Pinging again from triage: 
Have you had a chance to look at the review comments @rick68?

Thanks!

**Comment by rick68** [comment]
> Pinging again from triage:
> Have you had a chance to look at the review comments @rick68?
> 
> Thanks!

**Comment by rick68** [comment]
> Pinging again from triage:
> Have you had a chance to look at the review comments @rick68?
> 
> Thanks!

Hi @JohnCSimon, I had changed all comments in last PR.
Thanks for your help!

**Comment by Dylan-DPC** [comment]
looks good to me 

r? @Dylan-DPC @bors r+

**Pull request merged**